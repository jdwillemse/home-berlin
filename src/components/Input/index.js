import React from 'react';
import PropTypes from 'prop-types';

/**
 * Input component that can be resused for diffent input types.
 * It executes the form's checkValidity
 */
class Input extends React.PureComponent {
  /**
   * Render the component.
   * @return {node} - JSX node
   */
  render() {
    const { name, label, addRef, ...props } = this.props;

    return (
      <fieldset className="input__wrap clearfix" ref={addRef}>
        <label htmlFor={name}>
          <div className="input__label">{label}</div>
          <input
            {...props}
            id={name}
            name={name}
            className="input__field"
            aria-describedby={`${name}-error`}
            required
          />
          <div className="invalid-feedback" id={`${name}-error`} aria-live="polite" />
        </label>
      </fieldset>
    );
  }
}

Input.propTypes = {
  name: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
  addRef: PropTypes.func.isRequired
};

export default Input;
