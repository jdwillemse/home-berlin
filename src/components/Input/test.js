import React from 'react';

import Input from './index';

const inputProps = { onBlur: jest.fn(), addRef: jest.fn() };
const fields = [
  {
    name: 'phone',
    label: 'Phone number',
    type: 'tel',
    pattern: '^+?(d|-|s)+$'
  },
  {
    name: 'fullname',
    label: 'Full name',
    type: 'text'
  },
  {
    name: 'email',
    label: 'Email address',
    type: 'email'
  }
];

describe('Input', () => {
  beforeAll(() => {
    // Mock warn & error to catch errors thrown by React into the console
    console.warn = jest.fn();
    console.error = jest.fn();
  });

  afterEach(() => {
    expect(console.warn).not.toBeCalled();
    expect(console.error).not.toBeCalled();
  });

  it('should match snapshot', () => {
    for (let i = 0; i < fields.length; i++) {
      const wrapper = shallow(<Input {...fields[i]} {...inputProps} />);
      expect(toJson(wrapper)).toMatchSnapshot();
    }
  });
});
