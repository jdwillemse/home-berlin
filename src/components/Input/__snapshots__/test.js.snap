// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input should match snapshot 1`] = `
<fieldset
  className="input__wrap clearfix"
>
  <label
    htmlFor="phone"
  >
    <div
      className="input__label"
    >
      Phone number
    </div>
    <input
      aria-describedby="phone-error"
      className="input__field"
      id="phone"
      name="phone"
      onBlur={[MockFunction]}
      pattern="^+?(d|-|s)+$"
      required={true}
      type="tel"
    />
    <div
      aria-live="polite"
      className="invalid-feedback"
      id="phone-error"
    />
  </label>
</fieldset>
`;

exports[`Input should match snapshot 2`] = `
<fieldset
  className="input__wrap clearfix"
>
  <label
    htmlFor="fullname"
  >
    <div
      className="input__label"
    >
      Full name
    </div>
    <input
      aria-describedby="fullname-error"
      className="input__field"
      id="fullname"
      name="fullname"
      onBlur={[MockFunction]}
      required={true}
      type="text"
    />
    <div
      aria-live="polite"
      className="invalid-feedback"
      id="fullname-error"
    />
  </label>
</fieldset>
`;

exports[`Input should match snapshot 3`] = `
<fieldset
  className="input__wrap clearfix"
>
  <label
    htmlFor="email"
  >
    <div
      className="input__label"
    >
      Email address
    </div>
    <input
      aria-describedby="email-error"
      className="input__field"
      id="email"
      name="email"
      onBlur={[MockFunction]}
      required={true}
      type="email"
    />
    <div
      aria-live="polite"
      className="invalid-feedback"
      id="email-error"
    />
  </label>
</fieldset>
`;
